manifests/server/grant.pp

```diff
@@ -2,16 +2,31 @@
 define postgresql::server::grant (
   String $role,
   String $db,
-  Optional[String] $privilege      = undef,
-  String $object_type              = 'database',
-  Optional[String[1]] $object_name = undef,
-  String $psql_db                  = $postgresql::server::default_database,
-  String $psql_user                = $postgresql::server::user,
-  Integer $port                    = $postgresql::server::port,
-  Boolean $onlyif_exists           = false,
-  Hash $connect_settings           = $postgresql::server::default_connect_settings,
+  Optional[String] $privilege       = undef,
+  String $object_type               = 'database',
+  Optional[String[1]] $object_name  = undef,
+  String $psql_db                   = $postgresql::server::default_database,
+  String $psql_user                 = $postgresql::server::user,
+  Integer $port                     = $postgresql::server::port,
+  Boolean $onlyif_exists            = false,
+  Hash $connect_settings            = $postgresql::server::default_connect_settings,
+  Enum['present', 'absent'] $ensure = 'present',
 ) {
 
+  case $ensure {
+    'present': {
+      $sql_command = 'GRANT %s ON %s "%s" TO "%s"'
+      $unless_is = true
+    }
+    'absent': {
+      $sql_command = 'REVOKE %s ON %s "%s" FROM "%s"'
+      $unless_is = false
+    }
+    default: {
+      fail("Unknown value for ensure '${ensure}'.")
+    }
```

Comment:
I just copied this structure from grant_role.pp

I suspect it pre-dates the introduction of strict typing.

I will remove the redundant code.